### ISSUE TITLE
store.add({}).then(store.sync) fails

### ISSUE THREAD

**Issue by gr2m** [hoodiehq/hapi-couchdb-store#2]
```js
store.add({}).then(store.sync).catch(console.log.bind(console)); ''
```

fails with `400 Document must be a JSON object`

But this should work, as `store.sync` should accept an object as it's argument. This is most likely an issue with https://github.com/hoodiehq/pouchdb-hoodie-sync

**Comment by HipsterBrown** [comment]
I discovered that when `store.sync()` tries to resolve an object to an `id` in the `toId()` util, it looks for `docOrId._id` as the returned `id`. However, the object being passed into `store.sync()` has a the `id` stored as `docOrId.id`. 

So the question is, do we change `pouchdb-hoodie-sync` to look for an id under `._id` or `.id`? Or do we change `store.add()` to change the object being passed to `store.sync()`?

**Comment by HipsterBrown** [comment]
After modifying the `toId()` util to look for `.id` if `._id` was undefined, the server responded with an error during replication:

```json
{
   "statusCode":400,
   "error":"Bad Request",
   "message":"child \"path\" fails because [\"path\" with value \"0b8u2ld&#x2f;\" fails to match the database must start with character pattern]",
   "validation":{"source":"params","keys":["path"]}
}
```

The object passed into `store.sync()` resolved to `["F707D634-82D1-F251-B06A-31C4B844EC2D"]`. So I'm not quite sure what's happening.

**Comment by gr2m** [comment]
We actually have several issues here. The `400 bad Request` issue gets resolved here: https://github.com/hoodiehq/hapi-couchdb-store/pull/5

The original issue here remains though
